DragaliaAPI.Integration.Test/Features/Trade/TreasureTradeTest.cs

```diff
@@ -0,0 +1,124 @@
+
+public class TreasureTradeTest : TestFixture
+{
+    public TreasureTradeTest(
+        CustomWebApplicationFactory<Program> factory,
+        ITestOutputHelper outputHelper
+    )
+        : base(factory, outputHelper) { }
+
+    [Fact]
+    public async Task GetListAll_NoTrades_ReturnsEmpty()
+    {
+        this.ApiContext.PlayerTrades.RemoveRange(
+            this.ApiContext.PlayerTrades.Where(x => x.DeviceAccountId == DeviceAccountId)
+        );
+
+        await this.ApiContext.SaveChangesAsync();
+
+        TreasureTradeGetListAllData response = (
+            await Client.PostMsgpack<TreasureTradeGetListAllData>(
+                "treasure_trade/get_list_all",
+                new TreasureTradeGetListAllRequest()
+            )
+        ).data;
+
+        response.user_treasure_trade_list.Should().BeEmpty();
+        response.treasure_trade_all_list.Should().NotBeEmpty();
+        response.treasure_trade_list.Should().BeNullOrEmpty();
+    }
+
+    [Fact]
+    public async Task GetListAll_WithTrades_ReturnsTrades()
+    {
+        this.ApiContext.PlayerTrades.RemoveRange(
+            this.ApiContext.PlayerTrades.Where(x => x.DeviceAccountId == DeviceAccountId)
+        );
+
+        this.ApiContext.PlayerTrades.Add(
+            new DbPlayerTrade()
+            {
+                DeviceAccountId = DeviceAccountId,
+                Id = 1000,
+                Count = 1,
+                Type = TradeType.Treasure,
+                LastTradeTime = DateTimeOffset.UnixEpoch
+            }
+        );
+
+        await this.ApiContext.SaveChangesAsync();
+
+        TreasureTradeGetListAllData response = (
+            await Client.PostMsgpack<TreasureTradeGetListAllData>(
+                "treasure_trade/get_list_all",
+                new TreasureTradeGetListAllRequest()
+            )
+        ).data;
+
+        response.user_treasure_trade_list
+            .Should()
+            .HaveCount(1)
+            .And.ContainEquivalentOf(new UserTreasureTradeList(1000, 1, DateTimeOffset.UnixEpoch));
+        response.treasure_trade_all_list.Should().NotBeEmpty();
+        response.treasure_trade_list.Should().BeNullOrEmpty();
+    }
+
+    [Fact]
+    public async Task Trade_ValidTrade_Trades()
+    {
+        int preTradeAmount;
+
+        using (
+            IDisposable ctx = this.Services
+                .GetRequiredService<IPlayerIdentityService>()
+                .StartUserImpersonation(DeviceAccountId)
+        )
+        {
+            IInventoryRepository inv = this.Services.GetRequiredService<IInventoryRepository>();
+
+            DbPlayerMaterial mat =
+                await inv.GetMaterial(Materials.DamascusCrystal)
+                ?? inv.AddMaterial(Materials.DamascusCrystal);
+            mat.Quantity = 10;
+
+            preTradeAmount = (await inv.GetMaterial(Materials.DamascusIngot))?.Quantity ?? 0;
+
+            await this.Services.GetRequiredService<IUpdateDataService>().SaveChangesAsync();
+        }
```

Comment:
The test fixture already gives like 1k of every material I think, if that's what you wanted to do here